### DISCUSSION TITLE
Runtime error occurred in main trigger firing loop

### DISCUSSION THREAD

**Discussion by juriskur** [quartznet/quartznet#1238] [Q&A]
Hello,

I try to schedule and run pretty easy job without extra persistence/storage.
But instead of job running I got mentioned error message in log file and the job does not start.

Trigger/Job are created. No further error messages are given.

What could be reason for this problem?

Thanks in advance!

**Comment by juriskur** [comment]
ok, just found: the problem was the assembly: System.Diagnostics.DiagnosticSource presented in the wrong version in the project

**Reply by lahma** [reply]
I thought this was fixed in 3.3.2

**Reply by Tsiftelis-Thanasis** [reply]
Hello!
I am using the latest in a ASP.Net Webforms application with .Net 4.8 Framework.

I have the following errors:
**_[Error] Runtime error occurred in main trigger firing loop.
System.IO.FileNotFoundException: Could not load file or assembly 'System.Diagnostics.DiagnosticSource, Version=6.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.
File name: 'System.Diagnostics.DiagnosticSource, Version=6.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'_**

I had the System.Diagnostics.DiagnosticSource, I removed it and it only worked once.

**Reply by lahma** [reply]
Did you try another Quartz version?

**Reply by Tsiftelis-Thanasis** [reply]
The same with earlier versions and also with 3.3.2.

**Reply by lahma** [reply]
What about the latest version?

**Reply by Tsiftelis-Thanasis** [reply]
I was testing with the latest.

I've seen that a version earlier than 3, with no asynchronous calls, was working fine.

**Reply by Sicos1977** [reply]
Just use a binding redirect to fix this. First figure out what version of System.Diagnostics.DiagnosticSource is installed in your application and then add a binding redirect to this version.

**Reply by Tsiftelis-Thanasis** [reply]
Hello, the conflict is when we add the Azure.Core.
Azure.Core/1.37.0 has a dependency of System.Diagnostics.DiagnosticSource: 6.0.1 and Quartz/3.8.1 has System.Diagnostics.DiagnosticSource: 4.7.1.
